datafusion/functions/src/core/named_struct.rs

```diff
@@ -57,6 +57,15 @@ fn named_struct_expr(args: &[ColumnarValue]) -> Result<ColumnarValue> {
         .into_iter()
         .unzip();
 
+    // Check to enforce the uniqueness of struct field name
+    let unique_field_names_set = names.iter().collect::<HashSet<_>>();
+
+    if unique_field_names_set.len() != names.len() {
+        return exec_err!("named_struct requires unique field names");
+    }
+
+    drop(unique_field_names_set);
```

Comment:
`hashbrown::HashSet` holds the string references so dropping the set here.

Reply:
seems reasonable -- another way to do this is in a scope (enclosed with `{` and `}` 

```rust
{
  // Check to enforce the uniqueness of struct field name
    let unique_field_names_set = names.iter().collect::<HashSet<_>>();

    if unique_field_names_set.len() != names.len() {
        return exec_err!("named_struct requires unique field names");
}
```